units/src/amount/signed.rs

```diff
@@ -478,7 +510,7 @@ impl TryFrom<Amount> for SignedAmount {
 impl core::iter::Sum for SignedAmount {
     fn sum<I: Iterator<Item = Self>>(iter: I) -> Self {
         let sats: i64 = iter.map(|amt| amt.0).sum();
-        SignedAmount::from_sat(sats)
+        SignedAmount::from_sat_unchecked(sats)
```

Comment:
This screams if-we-forget-to-fix-this-we-will-release-buggy-code

I would suggest an additional PR be raised (built on top of this one) that enforces the invariant when summing.

Reply:
Good idea.  Although it wouldn't actually change anything in `sum()` but instead an updated to [checked_sum()](https://github.com/rust-bitcoin/rust-bitcoin/blob/61744ef568913b9c958a532ee1ca51afa11162b1/units/src/amount/mod.rs#L571) I believe (which apparently doesn't have a failing test case now that `SignedAmount::MAX` was changed).  I'm curious though why do we have `sum()` and `checked_sum()` in different places?

Reply:
After further inspection, if the invariant is enforced by `checked_add` then there's no need to update `checked_sum`

Reply:
Since we already plan to enforce this in arithmetic ops AFAIK marking this as resolved.